server/src/database/models/user.js

```diff
@@ -0,0 +1,74 @@
+
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import mongoose from 'mongoose';
+
+const { JWT_SECRET } = process.env;
+
+const userSchema = mongoose.Schema({
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  username: {
+    type: String,
+    required: true,
+    trim: true,
+    unique: true,
+    minlength: 3,
+  },
+  password: {
+    type: String,
+    required: true,
+    minlength: 6,
+  },
+});
+
+/**
+ * Overides the model's default JSON stringify method
+ * @returns Object with user ID and username
+ */
+userSchema.methods.toJSON = function toJSON() {
+  const user = this;
+  /* eslint-disable-next-line no-underscore-dangle */
+  return { _id: user._id, username: user.username };
+};
+
+/**
+ * Generates new JWT and returns it
+ * @returns JWT
+ */
+userSchema.methods.generateAuthToken = function generateAuthToken() {
+  const user = this;
+  const access = 'auth';
+  const token = jwt.sign(
+    {
+      /* eslint-disable-next-line no-underscore-dangle */
+      _id: user._id,
```

Comment:
Mongoose already adds id getter in the object, check [this](https://stackoverflow.com/a/34424834/10322565) for further info.
Also why `_id`?

Reply:
Okay, I will fix this.